internal/configs/harmony/harmony.go

```diff
@@ -0,0 +1,163 @@
+package harmony
+
+// HarmonyConfig contains all the configs user can set for running harmony binary. Served as the bridge
+// from user set flags to internal node configs. Also user can persist this structure to a toml file
+// to avoid inputting all arguments.
+type HarmonyConfig struct {
+	Version    string
+	General    GeneralConfig
+	Network    NetworkConfig
+	P2P        P2pConfig
+	HTTP       HttpConfig
+	WS         WsConfig
+	RPCOpt     RpcOptConfig
+	BLSKeys    BlsConfig
+	TxPool     TxPoolConfig
+	Pprof      PprofConfig
+	Log        LogConfig
+	Sync       SyncConfig
+	Sys        *SysConfig        `toml:",omitempty"`
+	Consensus  *ConsensusConfig  `toml:",omitempty"`
+	Devnet     *DevnetConfig     `toml:",omitempty"`
+	Revert     *RevertConfig     `toml:",omitempty"`
+	Legacy     *LegacyConfig     `toml:",omitempty"`
+	Prometheus *PrometheusConfig `toml:",omitempty"`
+	DNSSync    DnsSync
+}
+
+type DnsSync struct {
+	Port          int    // replaces: Network.DNSSyncPort
+	Zone          string // replaces: Network.DNSZone
+	LegacySyncing bool   // replaces: Network.LegacySyncing
+	Client        bool   // replaces: Sync.LegacyClient
+	Server        bool   // replaces: Sync.LegacyServer
+	ServerPort    int
+}
+
+type NetworkConfig struct {
+	NetworkType string
+	BootNodes   []string
+}
+
+type P2pConfig struct {
+	Port         int
+	IP           string
+	KeyFile      string
+	DHTDataStore *string `toml:",omitempty"`
+}
+
+type GeneralConfig struct {
+	NodeType         string
+	NoStaking        bool
+	ShardID          int
+	IsArchival       bool
+	IsBeaconArchival bool
+	IsOffline        bool
+	DataDir          string
+}
+
+type ConsensusConfig struct {
+	MinPeers     int
+	AggregateSig bool
+}
+
+type BlsConfig struct {
+	KeyDir   string
+	KeyFiles []string
+	MaxKeys  int
+
+	PassEnabled    bool
+	PassSrcType    string
+	PassFile       string
+	SavePassphrase bool
+
+	KMSEnabled       bool
+	KMSConfigSrcType string
+	KMSConfigFile    string
+}
+
+type TxPoolConfig struct {
+	BlacklistFile string
+}
+
+type PprofConfig struct {
+	Enabled    bool
+	ListenAddr string
+}
+
+type LogConfig struct {
+	Folder        string
+	FileName      string
+	RotateSize    int
+	Verbosity     int
+	VerbosePrints map[string]bool `toml:",omitempty"`
```

Comment:
@JackyWYX 

> 1. Since there is some change to `HarmonyConfig`, it will also have some change to the dumped toml file `harmony.conf`. Please increase the config version for the dump command, and add config migration rules for explicit version change.

Added migration from `2.0.0` to `2.1.0`
https://github.com/harmony-one/harmony/blob/b87f17f0effdcb2e186251e96df95390c1929fd1/cmd/harmony/config_migrations.go#L161-L169

> 2. What is the default value of this field? Shall we predefine some fields for user to config?

defaults:
```toml
[Log]
  [Log.VerbosePrints]
    Config = false
```

https://github.com/harmony-one/harmony/blob/b87f17f0effdcb2e186251e96df95390c1929fd1/cmd/harmony/default.go#L69-L77